graph-gateway/src/network/entities.rs

```diff
@@ -0,0 +1,192 @@
+//! Entities that are used to represent the network topology.
+
+use std::{
+    collections::{HashMap, HashSet},
+    fmt::Display,
+    ops::Deref,
+    sync::Arc,
+};
+
+pub use alloy_primitives::{Address, BlockNumber};
+use cost_model::CostModel;
+use custom_debug::CustomDebug;
+use eventuals::Ptr;
+use semver::Version;
+pub use thegraph_core::types::{DeploymentId, SubgraphId};
+use url::Url;
+
+/// The [`IndexingId`] struct represents the unique identifier of an indexing.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub struct IndexingId {
+    /// The indexer's ID indexing the subgraph's deployment.
+    pub indexer: Address,
+    /// The subgraph's deployment ID indexed by the indexer.
+    pub deployment: DeploymentId,
+}
+
+#[derive(Clone)]
+pub struct Indexing {
```

Comment:
I'm thinking that the term Indexing should be replaced with `Allocation`, and then we use the allocation id (`Address`) as keys in maps etc. And the `Allocation` struct would contain the corresponding indexer address, deployment id, etc.

Not necessary for this PR. Just a thought I had recently to remove this `Indexing` term that we've held over from the original TS gateway.

Reply:
Noted down for a future PR, because I think that would require we do the same with the whole codebase, e.g., the `client_query.rs`'s candidate preparation, which is the one that has conditioned this PR